### PR TITLE
DAOS-10525 pool: update dom status during reintegration.

### PR DESCRIPTION
**Pull request by wangdi1**
Only update domain status if the original status is DOWN
or DOWNOUT, i.e. partial targets reintegration may not
need change domain status.

Features: reintegration
Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

**Review by liuxuezhao — DISMISSED** [review]

**Review by jolivier23 — DISMISSED** [review]

**Comment by jolivier23** [comment]
should be run with Features: reintegration ?

**Comment by wangdi1** [comment]
> should be run with Features: reintegration ?

sure

**Review by liuxuezhao — APPROVED** [review]

File: src/pool/srv_pool_map.c [hunk 1]
```diff
@@ -126,9 +126,13 @@ update_one_tgt(struct pool_map *map, struct pool_target *target,
...
-				dom->do_comp.co_rank);
-			dom->do_comp.co_status = PO_COMP_ST_UP;
+			if (dom->do_comp.co_status == PO_COMP_ST_DOWNOUT ||
+			    dom->do_comp.co_status == PO_COMP_ST_DOWN) {
+				dom->do_comp.co_status = PO_COMP_ST_UP;
+				dom->do_comp.co_in_ver = target->ta_comp.co_in_ver;
+				D_DEBUG(DB_MD, "change rank %u to UP ver %u\n",
+					dom->do_comp.co_rank, dom->do_comp.co_in_ver);
+			}
 			dom->do_comp.co_flags = 0;
```
[the remark this answers is not in the document]
Comment (wangdi1) [reply src/pool/srv_pool_map.c:136]: I think the intention here is to change the domain (node) status to UP, once if there are any targets are UP due to the reintegration.

>  It seems strange that we clear all flags even if dom is not DOWNOUT or DOWN.

I think this is "wrong", we should only change it UP if it is down and downcut, or I miss your question.

[the remark this answers is not in the document]
Comment (liw) [reply src/pool/srv_pool_map.c:136]: Let me rephrase the question. For

    132 136 dom->do_comp.co_flags = 0;

do we really want to clear all flags in dom->do_comp.co_flags even if dom->do_comp.co_status is not DOWNOUT or DOWN?

[the remark this answers is not in the document]
Comment (liuxuezhao) [reply src/pool/srv_pool_map.c:136]: looks should set dom->do_comp.co_flags at same time-point as do_comp.co_status

[the remark this answers is not in the document]
Comment (wangdi1) [reply src/pool/srv_pool_map.c:136]: Hmm, I am not so sure about co_flags. But for now, it seems ok to set it to 0 in this case. 

Since domain status will not be DOWNOUT anymore, once there are any its targets become UP, then its co_flags can be set to 0, since DOWN2OUT only used for DOWNOUT target/domain.  @liuxuezhao ?

**Review by liw — APPROVED** [review]

File: src/pool/srv_pool_map.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (wangdi1) [reply src/pool/srv_pool_map.c:136]: >looks should set dom->do_comp.co_flags at same time-point as do_comp.co_status

will this cause issue if it only exclude partial targets from multiple ranks, then add those targets back, co_flags will not be set to 0 then?

[the remark this answers is not in the document]
Comment (liuxuezhao) [reply src/pool/srv_pool_map.c:136]: now the PO_COMPF_DOWN2OUT flag only checked when its status is PO_COMP_ST_DOWNOUT (to differenciate it is exlude or drain cause DOWNOUT), so seems the co_flags maybe can set together with co_status.

anyway, currently that is only used for target component, not used for domain type now. So either way should not cause logic error.
